pkg/restore/change_image_name_action.go

```diff
@@ -0,0 +1,280 @@
+					needUpdateObj = true
+					a.logger.Infof("Updating item's image from %s to %s", imageName, newImageName)
+					container.(map[string]interface{})["image"] = newImageName
+					initContainers[i] = container
+				}
+			}
+		}
+
+		if needUpdateObj {
+			if err := unstructured.SetNestedField(obj.UnstructuredContent(), initContainers, "spec", "jobTemplate", "spec", "template", "spec", "initContainers"); err != nil {
+				return nil, errors.Wrap(err, "unable to set item's initContainer image")
+			}
+		}
+
+	} else {
+		//handle containers
+		needUpdateObj := false
+		containers, _, err := unstructured.NestedSlice(obj.UnstructuredContent(), "spec", "template", "spec", "containers")
+		if err != nil {
+			a.logger.Infof("UnstructuredConverter meet error: %v", err)
+			return nil, errors.Wrap(err, "error getting item's spec.containers")
+		}
+		for i, container := range containers {
+			a.logger.Infoln("container:", container)
+			if image, ok := container.(map[string]interface{})["image"]; ok {
+				imageName := image.(string)
+				if exists, newImageName, err := a.isImageReplaceRuleExist(log, imageName, config); exists && err == nil {
+					needUpdateObj = true
+					a.logger.Infof("Updating item's image from %s to %s", imageName, newImageName)
+					container.(map[string]interface{})["image"] = newImageName
+					containers[i] = container
+				}
+			}
+		}
+
+		if needUpdateObj {
+			if err := unstructured.SetNestedField(obj.UnstructuredContent(), containers, "spec", "template", "spec", "containers"); err != nil {
+				return nil, errors.Wrap(err, "unable to set item's container image")
+			}
+			a.logger.Infof("obj.Object %v", obj.Object)
+		}
+
+		//handle initContainers
+		needUpdateObj = false
+		initContainers, _, err := unstructured.NestedSlice(obj.UnstructuredContent(), "spec", "template", "spec", "initContainers")
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting item's spec.initContainers")
+		}
+		for i, container := range initContainers {
+			if image, ok := container.(map[string]interface{})["image"]; ok {
+				imageName := image.(string)
+				if exists, newImageName, err := a.isImageReplaceRuleExist(log, imageName, config); exists && err == nil {
+					needUpdateObj = true
+					a.logger.Infof("Updating item's image from %s to %s", imageName, newImageName)
+					container.(map[string]interface{})["image"] = newImageName
+					initContainers[i] = container
+				}
+			}
+		}
+
+		if needUpdateObj {
+			if err := unstructured.SetNestedField(obj.UnstructuredContent(), initContainers, "spec", "template", "spec", "initContainers"); err != nil {
+				return nil, errors.Wrap(err, "unable to set item's initContainer image")
+			}
+		}
+
+	}
+	return velero.NewRestoreItemActionExecuteOutput(obj), nil
+}
+
+func (a *ChangeImageNameAction) isImageReplaceRuleExist(log *logrus.Entry, oldImageName string, cm *corev1.ConfigMap) (exists bool, newImageName string, err error) {
```

Comment:
Could we simplify the replace rule? The logic of this function is complicated and hard to understand.
And seems it doesn't handle all cases correctly, e.g.
The original image is `dev/image1:dev` `dev/image2:dev` and I want to replace the image with `test/image1:dev` `test/image2:dev`. If I use `dev,test` as the replace rule, I will get `test/image1:test` `test/image2:test`?

OMI, how about that we only support replacing the prefix of the image? This should cover most requirements and is easier.
e.g.
The registry IP address changed: `ip1/xxx` -> `ip2/xxx` with the replace rule `ip1,ip2`
The registry port changed: `ip1:5000` - > `ip1:6000` with the replace rule `ip1:5000,ip1:6000`
The image repository changed: `ip1/repo1/xxx` -> `ip1/repo2/xxx` with the replace rule `ip1/repo1,ip1/repo2`
@wenterjoy @sseago WDYT?

Reply:
@ywk253100 
Thanks a lot for your comments and suggestion.
I do agree with you to cover most requirements. 
But if we support more cases, the code logic would be more complex.
e.g. ip1:5000 - > ip1:6000   ip1:tag1 -> ip1:tag2
should we distinguish between port number and tag？
current implement could cover these cases with replacing the keywords, but it indeed not user-friendly. 
e.g. dev/image1:dev -> test/image1:dev   dev/image2:dev -> test/image2:dev 
if users want to only replace first "dev", the replace rule should be  "dev/, test/". 
This will leave users to be responsible for their wants. I not sure it is in line with velero's principles.